### PR TITLE
Support arbitrary -gX debug options, and limit DWARF to line-tables-only

### LINKED ISSUE #9611: support `-gline-tables-only`
`-gline-tables-only` is a useful if somewhat obscure flag in clang which causes the compiler to generate line table debug info but not full info (e.g. source variables). It's useful for allowing e.g. symbolized stack traces without having all of the debug info (which is nice in large programs). It also happens to fit quite well with the current status quo of web debugging which is source-maps-only, and the fact that binaries bloated with huge debug info are more annoying in wasm (even for local debugging) than native (since loading the binary is more costly than loading a native binary). So it would be nice to at least pass the option through to clang; it probably also makes sense to preserve the name section in this case.

### PR DESCRIPTION
**Pull request by kripken**
Fixes #9611

Sadly this adds some complexity, but I tried hard to keep
it all in one area, and not spread around.

### CODE REVIEW

**Comment by dschuff** [comment]
After looking at dwarf-splitting, I think we're going to want to make at least the compile-part of it a fully general passthrough, because the flag for splitting is `-gsplit-dwarf[=single]`

**Comment by dschuff** [comment]
This is also weird because  `g1` is more debug info than line-tables-only but we are modeling that as level 3.
Also we may want the ability to control generation of source maps independently from names sections and dwarf sections.
Maybe we need to make a complete separation between flags used at compile time and link time, where we pass through at compile times and accept a different set of flags at link time. It's probably already the case that users  are doing special things for link anyway because build systems don't pass `-g` at link time by default.

**Comment by kripken** [comment]
Added arbitrary passthrough support, and made it limit debug info to what we can do for now, which is line tables.

File: emcc.py [hunk 1]
```diff
@@ -2640,10 +2638,26 @@ def check_bad_eq(arg):
...
-      options.debug_level = validate_arg_level(requested_level, 4, 'Invalid debug level: ' + newargs[i])
       options.requested_debug = newargs[i]
-      newargs[i] = ''
+      requested_level = newargs[i][2:] or '3'
+      if is_int(requested_level):
+        # the -gX value is the debug level (-g1, -g2, etc.)
+        options.debug_level = validate_arg_level(requested_level, 4, 'Invalid debug level: ' + newargs[i])
+        # if we don't need to preserve LLVM debug info, do not keep this flag
+        # for clang
+        if options.debug_level < 3:
```
Comment (dschuff) [inline emcc.py]: Doesn't this mean that you can never get `g1`-level debug info from the frontend?

Comment (kripken) [reply emcc.py]: Oh, does `-g1` mean something useful for clang? `clang --help` doesn't show anything.

**Comment by dschuff** [comment]
Clang's options are not well documented. But the behavior tends to match
https://gcc.gnu.org/onlinedocs/gcc-7.5.0/gcc/Debugging-Options.html#Debugging-Opti

**Comment by kripken** [comment]
I see. Then yes, `-g1` in emcc is different from gcc/clang, and we've not supported the gcc/clang meaning. This PR doesn't change that, but yeah, that's unfortunate.

**Comment by kripken** [comment]
True. Maybe that refactoring could be helped by @sbc100's planned refactoring of an emlink type tool, which separates the linking part completely.

**Pull request merged**